bin/retrieve_dns.py

```diff
@@ -114,14 +114,17 @@ def get_xml(url, proxy):
     '''
     try:
         # Try without a proxy
-        conn = urllib.urlopen(url)
+        conn = urllib.request.urlopen(url)
         dn_xml = conn.read()
         conn.close()
-    except IOError:
+        # TODO: If we are the owners of the url specified then we can catch proper errors than all.
+    except Exception:
```

Comment:
We probably want to catch more specific exceptions here?

Reply:
`URLError` reads to me as the base class for all `urllib.request` exceptions: https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError, which would be better to catch then just `Exception`.

> The [urllib.error](https://docs.python.org/3/library/urllib.error.html#module-urllib.error) module defines the exception classes for exceptions raised by [urllib.request](https://docs.python.org/3/library/urllib.request.html#module-urllib.request). The base exception class is [URLError](https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError).

Reply:
I have added this in https://github.com/apel/apel/pull/354 https://github.com/apel/apel/commit/5b581a25e47958f5cd9a867dcb80609d4ab72d82. Just to acknowledge, Greg